src/pages/2023-12/next-js-app-router-cache/index.mdx

```diff
@@ -0,0 +1,388 @@
+2. **Cache Miss**: If this is the first time the page is being accessed or the cache has expired, i.e. revalidated, it results in a cache miss.
+3. **Fetching the data:** Next, we need to fetch the data necessary to render our page. First, we check the data cache. If not found in the data cache, it is fetched from the data source.
+4. **Rendering to Payload:** The fetched data is used to render the RSC payload, containing the necessary information to render the server components and their interaction with the client components.
+5. **Hydrating the HTML:** The React Server payload will hydrate the HTML, creating interactivity between the server-side and the client-side.
+6. **Cache set:** Now that the HTML and the RSC payload is fetched, it is cached inside of the full route cache, resulting in faster subsequent page loads.
+7. **Subsequent Requests**: If the cache has not expired by the time another request is made, it will result in a cache "HIT", enabling quick hydration of the page on client side.
+
+From the description above, I am sure you are able to see the direct interaction between the full route cache and the data cache. This goes to prove the point I made earlier about the caching mechanisms working together in unison.
+
+The caching of the route is dependent on its nature. If a route is static, something like a blog, it's cached by default. It doesn't make sense to make redundant requests to the server for static content.
+
+Full route cache is skipped for dynamic routes.
+
+<Tangent>
+
+React Server Component Payload: Think of it like a toolbox that contains all the instructors and rules for how the server-rendered components should integrate and interact with the client-side components. It also contains placeholders for client components.
+Placeholders indicate where the client components fit into the overall structure of the page.
+
+</Tangent>
+
+```jsx {5}
+import Link from "next/link";
+
+async function getBlogList() {
+  const blogPosts = await fetch("https://api.blogposts.com/1", {
+    cache: "force-cache",
+  });
+
+  const blogData = await blogPosts.json();
+  return blogData;
+}
+
+export default async function Page() {
+  const blogData = await getBlogList();
+
+  return (
+    <div>
+      <h1>Blog Posts</h1>
+      <ul>
+        {blogData.map((post) => (
+          <li key={post.slug}>
+            <Link href={`/blog/${post.slug}`}>
+              <a>{post.title}</a>
+            </Link>
+            <p>{post.excerpt}</p>
+          </li>
+        ))}
+      </ul>
+    </div>
+  );
+}
+```
+
+In the above code, `blogData` returns static content, which is cached by default. I have put a `{cache: "force-cache"}` for illustrative purposes but it can be ommited.
+At build time, the RSC Payload and HTML will be cached in the Full Route Cache. This means that when the user runs the application at run-time to access `/page`, they will be served from the Full Route Cache, resulting in faster page loads.
+
+##### Opting out
```

Comment:
This never mentions how this works with dynamic functions, or dynamic URL params.